clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1307,11 +1316,13 @@ public void seek(TopicPartition partition, long offset) {
      * Seek to the first offset for each of the given partitions. This function evaluates lazily, seeking to the
      * first offset in all partitions only when {@link #poll(long)} or {@link #position(TopicPartition)} are called.
      * If no partition is provided, seek to the first offset for all of the currently assigned partitions.
+     *
+     * @throws IllegalArgumentException if provided TopicPartition is not assigned to this consumer
      */
     public void seekToBeginning(Collection<TopicPartition> partitions) {
         acquireAndEnsureOpen();
         try {
-            Collection<TopicPartition> parts = partitions.size() == 0 ? this.subscriptions.assignedPartitions() : partitions;
+            Collection<TopicPartition> parts = partitions == null || partitions.size() == 0 ? this.subscriptions.assignedPartitions() : partitions;
```

Comment:
Hmm.. should we enforce `partitions` to be not null actually? @hachikuji

Reply:
Huh, weird. Didn't realize we implemented this behavior. Seems like a better way would have been to have a no-arg `seekToBeginning()`. I think I'm with @guozhangwang. Maybe we just raise an exception on null? This matches current behavior.

Reply:
what about `size() == 0` ?

Reply:
We probably have to keep the `size() == 0` behavior for compatibility.

Reply:
Never mind. JavaDocs explains that `size() == 0` is supported.